### PR TITLE
navigator key added

### PR DESCRIPTION
**Pull request by xoltawn**
It is related to this issue:
https://github.com/tomgilder/routemaster/issues/195
I sent a pull request earlier but it has conflicts with main branch. so I send another PR regarding that issue
(sorry for the delay)

### CODE REVIEW

**Comment by xoltawn** [comment]
you just need to add 
```navigatorKey: GlobalKey<NavigatorState>(),```
to your 
```RoutemasterDelegate```;

**Comment by anthonysette** [comment]
This would be a great feature. I was reviewing your code, why did you import lib/app.dart? @tomgilder could you review this? 

My use case would be in the situation where you want to send in-app notifications. These snack bars depend on the current context and as of now, it needs to be passed around to keep it up to date with the current page. In some situations, the context is not current and so we get an error looking up a deactivated widget. If we had access to the current context for any page it would be a lot simpler to build out a notification handler. And there are plenty of other use cases such as the one @xoltawn mentioned in his last PR. 

Let's get this pushed through, this is a great addition to the package.

**Comment by behzadcym** [comment]
@anthonysette 
I guess importing lib/app.dart it was an accident. I'm gonna remove it push again

**Comment by anthonysette** [comment]
That would be great! We need to get a reviewer in here to close the PR. Once the fix is up I will test it, but I love this idea!

**Comment by xoltawn** [comment]
@anthonysette I just don't understand. I run ```flutter test``` on my machine and all tests pass. how should I check for more errors?

**Comment by anthonysette** [comment]
@tomgilder Can we merge this? The community would benefit a lot from this!

**Review by anthonysette — APPROVED** [review]
I have tested and am currently using these changes in production. This PR is solid. There have been no issues.

**Comment by xoltawn** [comment]
@tomgilder Will this be merged?

File: lib/routemaster.dart [hunk 1]
```diff
@@ -1048,7 +1057,14 @@ class RoutemasterDelegate extends RouterDelegate<RouteData>
       completer.complete(route);
     }
   }
+  // /// it will be passed to PageStackNavigator() which is going to be used as 'key' for Navigator()
+  // final GlobalKey<NavigatorState>? _navigatorKey = GlobalKey<NavigatorState>();
+  //
+  // @override
+  // GlobalKey<NavigatorState>? get navigatorKey => _navigatorKey;
```
Comment (BazinC) [inline lib/routemaster.dart:1064]: please remove if not necessary

File: lib/routemaster.dart [hunk 2]
```diff
@@ -313,8 +318,10 @@ class RoutemasterDelegate extends RouterDelegate<RouteData>
     required this.routesBuilder,
     this.transitionDelegate,
     this.observers = const [],
+    this.navigatorKey
   }) : navigatorBuilder = null {
     _state.delegate = this;
+    // _navigatorKey = GlobalKey<NavigatorState>();
```
Comment (BazinC) [inline lib/routemaster.dart:324]: remove if not necessary

File: lib/routemaster.dart [hunk 3]
```diff
@@ -224,9 +227,9 @@ class Routemaster {
   ///
   @optionalTypeArgs
   NavigationResult<T> push<T extends Object?>(
-    String path, {
-    Map<String, String>? queryParameters,
-  }) {
+      String path, {
+        Map<String, String>? queryParameters,
+      }) {
```
Comment (BazinC) [inline lib/routemaster.dart:232]: hey @xoltawn , I think you don't use the same dart formatter as @tomgilder , which results in a lot of unnecessary changes :)

File: lib/routemaster.dart (hunk 2 above)
Comment (xoltawn) [reply lib/routemaster.dart]: I removed them

File: lib/routemaster.dart (hunk 3 above)
Comment (xoltawn) [reply lib/routemaster.dart]: I edited the files and now we have the lease changes possible.

**Comment by xoltawn** [comment]
@BazinC changes are made to solve the issues you mentioned

**Comment by tomgilder** [comment]
Hi @xoltawn @anthonysette I've added this a slightly different way in PR #262.

Providing a `GlobalKey<NavigatorState>` will give you access to the context and navigator:

```dart
final navigatorKey = GlobalKey<NavigatorState>();

final delegate = RoutemasterDelegate(
  navigatorKey: navigatorKey,
  routesBuilder: (_) => RouteMap(
    routes: {
      '/': (_) => const MaterialPageOne(),
      '/two': (_) => const MaterialPageTwo(),
    },
  ),
);

// Get the current context
final context = navigatorKey.currentContext!;

// Pop the navigator
navigatorKey.currentState!.pop();
```
